src/pipecat/services/ai_services.py

```diff
@@ -227,6 +237,22 @@ async def process_frame(self, frame: Frame, direction: FrameDirection):
         else:
             await self.push_frame(frame, direction)
 
+    async def push_frame(self, frame: Frame, direction: FrameDirection = FrameDirection.DOWNSTREAM):
+        await super().push_frame(frame, direction)
+
+        if self._stop_frame_task is not None:
+            # Reschedule timeout task if it was already running
+            self._stop_frame_task.cancel()
+            self._stop_frame_task = self.get_event_loop().create_task(self._stop_frame_handler())
```

Comment:
I believe this will generate lots of task creation. There are services that produce thousands of `AudioRawFrame`s. Also, we would only do this if `isinatance(frame, AudioRawFrame)`.

Just talking out loud, but I was thinking that maybe we could just use one task and a queue with a `async.wait_for(queue.get(), timeout)`.

Something like this

```
if isinstance(frame, StartInterruptionFrame) or isinstance(frame, TTSStartedFrame) or  isinstance(frame, AudioRawFrame):
  await self._tts_stop_queue.put(frame)
```

then

```
async def _stop_frame_handler(self):
    tts_start_frame_seen = False
    while True:
        try:
            frame = await asyncio.wait_for(self._tts_stop_queue.get(), self._stop_frame_timeout)
            if isinstance(frame, StartInterruptionFrame):
                    tts_start_frame_seen = False
            elif isinstance(frame, TTSStartFrame):
                    tts_start_frame_seen = True
        except asyncio.TimeoutError:
            if tts_start_frame_seen:
                await self.push_frame(TTSStoppedFrame())
                tts_start_frame_seen = False
```

Maybe? Totally untested...